subprojects/language-scala/src/main/java/org/gradle/api/internal/tasks/scala/NormalizingScalaCompiler.java

```diff
@@ -43,13 +45,18 @@ public NormalizingScalaCompiler(Compiler<ScalaJavaJointCompileSpec> delegate) {
 
     @Override
     public WorkResult execute(ScalaJavaJointCompileSpec spec) {
+        resolveSourceFiles(spec);
         resolveClasspath(spec);
         resolveNonStringsInCompilerArgs(spec);
         logSourceFiles(spec);
         logCompilerArguments(spec);
         return delegateAndHandleErrors(spec);
     }
 
+    private void resolveSourceFiles(JavaCompileSpec spec) {
+        spec.setSourceFiles(ImmutableSet.copyOf(spec.getSourceFiles()));
+    }
```

Comment:
This is just ugly... :/

Reply:
Well, it is basically the same code we had before...

Reply:
Only that now we use an `ImmutableSet` instead of `SimpleFileCollection`.